test/tests.test.ts

```diff
@@ -111,11 +120,8 @@ describe('build', () => {
         test.concurrent(name, async () => {
           const expected =
             src === null ? null : await loadFileContent(src, encoding)
-          const actual =
-            encoding === 'buffer'
-              ? await fetchBufferContent(server, dest)
-              : await fetchTextContent(server, dest)
-          expect(actual).toStrictEqual(transformedContent ?? expected)
+          const actual = await fetchContent(server, dest, encoding)
+          expect(actual.content).toStrictEqual(transformedContent ?? expected)
```

Comment:
```suggestion
          const actual = await fetchContent(server, dest, encoding)
          expect(actual.content).toStrictEqual(transformedContent ?? expected)

          if (contentType !== undefined) {
            expect(actual.contentType).toStrictEqual(contentType)
          }
```
I'd like this test to pass to make the behavior consistent with the preview server. In other words, I'd like the `Content-Type` header to be only inferred from the path extension.